src/Microsoft.OData.Core/ODataBatchOperationRequestMessage.cs

```diff
@@ -99,6 +121,31 @@ public string Method
         /// </summary>
         public IServiceProvider Container { get; private set; }
 
+        /// <summary>
+        /// The Group Id for this request message. Can be null.
+        /// </summary>
+        /// <returns>The Group Id for this request message.</returns>
+        public string GroupId
+        {
+            get { return this.groupId; }
+        }
+
+        /// <summary>
+        /// Gets the prerequisite request or group ids.
+        /// </summary>
+        internal IEnumerable<string> DependsOnIds
+        {
+            get
+            {
+                return this.dependsOnIds;
+            }
+
+            set
+            {
+                this.dependsOnIds = value;
+            }

```

Comment:
Rather than expose a setter for an IEnumerable, if we expose as a read-only property of type List<string> the user can add dependency Ids rather than replacing the entire enumerable..